index.html

```diff
@@ -207,256 +242,549 @@ <h2>Les tableaux</h2>
+Et le javascript c'était pas mieux avant!`;
+							</code>
+						</pre>
+				</section>
+			</section>
+			<section id="part3">
+				<section>
+					<h1>Programmation en JS</h1>
+				</section>
+				<section>
+					<h1>Les fonctions</h1>
+					<p>Les fonctions sont un peu la base de toute programmation... Le Javascript n'échappe pas à la règle.</p>
+					<p>En JavaScript, les fonctions sont des objets de première classe. Cela signifie qu'elles peuvent être manipulées et échangées, qu'elles peuvent avoir des propriétés et des méthodes, comme tous les autres objets JavaScript. Les fonctions sont des objets <a href="https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Fonctions">Function</a></p>
+					<p>Afin de renvoyer une valeur, la fonction doit comporter une instruction return. Une fonction qui ne renvoie pas de valeur retourne undefined.</p>
 				</section>
 				<section>
-					<h2>Pour éviter les écueils, utiliser ===</h2>
+					<h1>Déclaration</h1>
 					<pre>
-						<code>
-// avec ===, fini de jouer : vérif valeur ET TYPE !
-
-42 === '42'        // => false
-null === undefined // => false
-null === 0         // => false
-0 === undefined    // => false
-0 === false        // => false
-'0' === false      // => false
-NaN === NaN        // => false -- rien à faire !
+						<code class="javascript">
+/// Fonction standard
+function a() {
+	/// this = contexte appelant
+}
+/// Variable fonction anonyme (ne pas utiliser)
+const b = function() {
+	/// this = contexte appelant
+}
+/// Variable fonction nommée (usage assez rare)
+const c = function c() {
+	/// this = contexte appelant
+}
+/// Lambda / "fat arrow" (anonyme)
+const d = () => {
+	/// this = contexte de la déclaration
+}
+/// fonction auto-appelante (usage particulier)
+(function(){
+	/// this = contexte appelant
+})()
 						</code>
 					</pre>
 				</section>
 				<section>
-					<h2>in et delete</h2>
-					<pre>
-						<code class='javascript'>
+						<h1>Déclaration</h1>
+						<pre>
+							<code class="javascript">
+/// La syntaxe de a est un raccourci pour celle de b. (usage assez rare)
+const e {
+    a() {
+        /// this = contexte appelant
+    },
+    b: function b() {
+        /// this = contexte appelant
+    }
+}
+/// Dans une classe
+class G {
+    a() {
+        /// this = contexte appelant
+		}
+		/// Variable d'instance
+    b = () => {
+        /// this = instance de la classe
+    }
+}
+							</code>
+						</pre>
+					</section>
+				<section>
+					<h1>Variables</h1>
+					<p>Il existe trois façons de déclarer des variable en javascript:</p>
+					<ul>
+						<li>var: On déclare une variable, éventuellement en initialisant sa valeur.</li>
+						<li>let: On déclare une variable dont la portée est celle du bloc courant, éventuellement en initialisant sa valeur.</li>
+						<li>const: On déclare une constante nommée, dont la portée est celle du bloc courant, accessible en lecture seule.</li>
```

Comment:
Le fait de parler ici d'un seul coup de "constante" et en fait surout le terme "nommé" me fait bizarre.

```suggestion
						<li>const: On déclare une variable, dont la portée est celle du bloc courant, accessible en lecture seule.</li>
```

Reply:
Pour le coup^s'est tiré directement de la doc MDN.

Le changé en variable ne me dérange pas. Mais j'aimais bien appuyer sur le côté constant de la valeur